pages/api/auth/[...nextauth].ts

```diff
@@ -1,70 +1,135 @@
-      try {
-        if (!profile?.login) return "/?authError=Profile not found";
-
-        return true;
-      } catch(e) {
-        console.log("SignIn Callback", e);
-        return false;
-      }
-    },
-    async jwt({ token, account, profile }) {
-      const user = await models.user.findOne({
-        where: { 
-          githubLogin: profile?.login || token?.login
-        },
-        raw: true
-      })
-        .catch(error => {
-          console.log("JWT Callback", error)
-        });
-
-      return { ...token, ...profile, ...account, wallet: user?.address };
-    },
-    async session({ session, token }) {
-      // console.log(`Session`, session, user, token);
-      return {
-        expires: session.expires,
-        user: {
-          ...session.user,
-          login: token.login,
-          accessToken: token?.access_token,
-        },
-      };
-    },
-  },
-});
+import { NextApiRequest, NextApiResponse } from "next";
+import NextAuth, { Account, Profile } from "next-auth";
+import { getToken } from "next-auth/jwt";
+import getConfig from "next/config";
+
+import models from "db/models";
+
+import { DAY_IN_SECONDS } from "helpers/constants";
+import { caseInsensitiveEqual } from "helpers/db/conditionals";
+import { toLower } from "helpers/string";
+
+import { Logger } from "services/logging";
+
+import { EthereumProvider, GHProvider } from "server/auth/providers";
+
+const {
+  publicRuntimeConfig,
+  serverRuntimeConfig: {
+    auth: {
+      secret
+    }
+  }
+} = getConfig();
+
+export default async function auth(req: NextApiRequest, res: NextApiResponse) {
+  const currentToken = await getToken({ req, secret: secret });
+
+  return NextAuth(req, res, {
+    providers: [
+      EthereumProvider,
+      GHProvider
+    ],
+    pages: {
+      signIn: "/auth/signin"
+    },
+    session:{
+      strategy: "jwt",
+      maxAge: 30 * DAY_IN_SECONDS
+    },
+    callbacks: {
+      async signIn({ profile, account}) {
+        try {
+          const provider = account?.provider;
+
+          if (provider === "github") {
+            if (!profile?.login) return "/?authError=Profile not found";
+
+            return true;
+          } else if (provider === "credentials") {
+            const accountAddress = account?.providerAccountId;
+
+            if (accountAddress) {
+              await models.user.findOrCreate({
+                where: {
+                  address: caseInsensitiveEqual("address", accountAddress.toLowerCase())
+                },
+                defaults: {
+                  address: accountAddress
+                }
+              });
+  
+              return true;
+            }
+          }
+        } catch(error) {
+          Logger.error(error, "SignIn callback: ");
+        }
+
+        return false;
+      },
+      async jwt({ token, profile, account }) {
+        try {
+          const provider = account?.provider;
+          
+          if (provider === "github") {

```

Comment:
this can be a `switch {}` instead, and we can export the logic on the else condition to a file because both share the return logic (object is same) where the only difference is the conditional logic (aka where information is stored)